### PR TITLE
[stdlib] Fix UnsafeBufferPointer range subscript getter tests

### PR DESCRIPTION
**Pull request by austinzheng**
#### What's in this pull request?

Fixing an issue with a unit test that caused its four variations to not all be run. Sorry! @gribozavr @moiseev 

OS X tests pass.

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

All four of the Unsafe*BufferPointer range subscript getter tests (one for each range) were using the same test name string. This caused only one of the four tests to actually be registered to the test suite. This change fixes that issue, and adds a couple more tests for better coverage.

### CODE REVIEW

**Comment by austinzheng** [comment]
@gribozavr I'm not in any rush to get this PR merged, but if it is merged I have a follow-up PR with more tests that depends on this one. (I'm also still working on the hashed collections PR.)

**Comment by moiseev** [comment]
@swift-ci Please test

**Comment by gribozavr** [comment]
LGTM.

File: validation-test/stdlib/UnsafeBufferPointer.swift.gyb [hunk 1]
```diff
@@ -15,6 +15,8 @@ struct SubscriptGetTest {
   /// The values that should be expected by slicing the UBP, or `nil` if the
   /// test is expected to crash.
   let expectedValues: [Int]?
+  /// Same as `expectedValues`, but for closed ranges. `nil` if no difference.
+  let expectedClosedValues: [Int]?
```
Comment (gribozavr) [inline validation-test/stdlib/UnsafeBufferPointer.swift.gyb]: There is `let subscriptRangeTests` in StdlibCollectionUnittest, which is very similar in intent.

Comment (austinzheng) [inline validation-test/stdlib/UnsafeBufferPointer.swift.gyb]: Those tests don't seem to be instrumented for use with the different types of ranges. Would a follow-up item to fix that be of interest to you?

Comment (gribozavr) [inline validation-test/stdlib/UnsafeBufferPointer.swift.gyb]: Absolutely, having more powerful shared datasets would make all tests stronger.

**Pull request merged**